src/main/java/com/mraof/minestuck/entity/ai/attack/FireballShootGoal.java

```diff
@@ -0,0 +1,123 @@
+package com.mraof.minestuck.entity.ai.attack;
+
+import com.mraof.minestuck.entity.AttackingAnimatedEntity;
+import com.mraof.minestuck.entity.animation.PhasedMobAnimation;
+import net.minecraft.core.BlockPos;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.PathfinderMob;
+import net.minecraft.world.entity.projectile.SmallFireball;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.Vec3;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Alternative attack for when the attack target is out of standard range, a small fireball is shot at the targets initial position.
+ */
+public class FireballShootGoal<T extends AttackingAnimatedEntity> extends AnimatedAttackWhenInRangeGoal<T>
+{
+	public static final int GROUP_SHOOT_COOLDOWN = 20; //how long it should take before another nearby mob can shoot fireballs
+	
+	private BlockPos initialTargetPos;
```

Comment:
Doesn't look like this pos is used properly. Is it meant to be used in place of the target position in `attemptToLandAttack()`?

Reply:
Alright, you'll need to fix the goal so that's the case then

Reply:
oh I see now hehe its not actually used